### PR TITLE
Log aggregation

### PR DESCRIPTION
**Pull request by osahyoun**
function, script and configuration for subscribing log groups to a lambda function and then on to papertrail.

### CODE REVIEW

**Comment by osahyoun** [comment]
Ping!

File: settings/production.yml [hunk 1]
```diff
@@ -14,3 +14,5 @@ environment:
   UNSUBSCRIBE_PAGE_NAME: ${ssm:/api-services/production/UNSUBSCRIBE_PAGE_NAME}
   BRAINTREE_MERCHANT_CURRENCIES: ${ssm:/api-services/production/BRAINTREE_MERCHANT_CURRENCIES}
   COGNITO_POOL_ARN: ${ssm:/api-services/production/COGNITO_POOL_ARN}
+  PAPERTRAIL_HOST: ${ssm:/api-services/production/PAPERTRAIL_HOST}
+  PAPERTRAIL_PORT: 34848
```
Comment (eyko) [inline settings/production.yml]: Why hardcoded here?

Comment (osahyoun) [reply settings/production.yml]: A very good question

File: scripts/subscribe-logs.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Script will create a stream subscription between a log group
+and a lambda function (see arn, below).
+
+ENV vars for ACCOUNT_ID and REGION are required.
+*/
+
+// ACCOUNT_ID=xxxxxxxxx STAGE=omar npm run
+
+const AWS = require('aws-sdk');
+const cloudwatchlogs = new AWS.CloudWatchLogs({ region: 'us-east-1' });
```
Comment (rodrei) [inline scripts/subscribe-logs.js]: Should `us-east-1` be in an ENV var?

File: scripts/subscribe-logs.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    cloudwatchlogs
+      .putSubscriptionFilter(options)
+      .promise()
+      .then(data => {
+        console.log(data);
+      })
+      .catch(err => {
+        console.log(err);
+      });
+  }, 1000);
```
Comment (rodrei) [inline scripts/subscribe-logs.js]: Can you please comment on why are you doing thins inside a `setTimeout` call?

File: scripts/subscribe-logs.js (hunk 2 above)
Comment (eyko) [reply scripts/subscribe-logs.js]: +1 for env-var, I like using `eu-west-2` for my stack

File: scripts/subscribe-logs.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+const AWS = require('aws-sdk');
+const cloudwatchlogs = new AWS.CloudWatchLogs({ region: 'us-east-1' });
+const accountId = process.env.ACCOUNT_ID;
+const stage = process.env.STAGE;
+const arn = `arn:aws:lambda:us-east-1:${accountId}:function:api-services-${
+  stage
+}-log-streamer`;
+
+const subscribe = item => {
```
Comment (rodrei) [inline scripts/subscribe-logs.js]: I don't understand what is `item` in this context

File: scripts/subscribe-logs.js (hunk 2 above)
Comment (osahyoun) [reply scripts/subscribe-logs.js]: 👍

**Comment by rodrei** [comment]
@osahyoun, can you please explain a little what you have done here? It would also be helpful to have an example of how this would be used. I'm guessing we need to use the `subscribe` function from the endpoints we want to log.

File: scripts/subscribe-logs.js (hunk 3 above)
Comment (osahyoun) [reply scripts/subscribe-logs.js]: So as to not exceed AWS API request limits

Comment (eyko) [reply scripts/subscribe-logs.js]: In that case wouldn't you want to schedule it for "a second after the last one"? It seems like you'll simply delay all logGroup subscription calls by a second, but you'll still call them all at once.

Comment (osahyoun) [reply scripts/subscribe-logs.js]: yes, indeed. I'd meant to multiply it by iteration index.

**Comment by rodrei** [comment]
hi @osahyoun! 👋

**Pull request closed without merging**